File: viewpage2_add_tablayout_tojava/app/src/main/java/com/example/viewpage2_add_tablayout_tojava/MainActivity.java
package com.example.viewpage2_add_tablayout_tojava;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.viewpager2.widget.ViewPager2;

import com.google.android.material.tabs.TabLayout;

public class MainActivity extends AppCompatActivity {

    private ViewPager2 viewPager;
    private TabLayout tabLayout;
    private FragmentViewPagerAdapter adapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        tabLayout = findViewById(R.id.tabLayout);
        viewPager = findViewById(R.id.viewpager);

        //創建一個adapter
        adapter = new FragmentViewPagerAdapter(this);

        //將 adapter 放入 ViewPager
        viewPager.setAdapter(adapter);

        //指定一開始 ViewPager 顯示內容位置
        viewPager.setCurrentItem(0);

        String[] title = new String[]{"今天", "明天", "未來七天"};
        tabLayout.addTab(tabLayout.newTab().setText(title[0]));
        tabLayout.addTab(tabLayout.newTab().setText(title[1]));
        tabLayout.addTab(tabLayout.newTab().setText(title[2]));

        //tabLayout 查詢事件
        tabLayout.addOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
            @Override
            public void onTabSelected(TabLayout.Tab tab) {
                viewPager.setCurrentItem(tab.getPosition());
            }

            @Override
            public void onTabUnselected(TabLayout.Tab tab) {

            }

            @Override
            public void onTabReselected(TabLayout.Tab tab) {

            }
        });
    }
}

File: ViewPager2_toKotlin/app/src/main/java/com/example/viewpager2_tokotlin/MainActivity.kt
package com.example.viewpager2_tokotlin

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.viewpager2.widget.ViewPager2

class MainActivity : AppCompatActivity() {

    private var viewPager: ViewPager2? = null
    private var adapter: FragmentViewPagerAdapter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        viewPager = findViewById(R.id.viewpager)

        //創建一個adapter
        adapter = FragmentViewPagerAdapter(this)

        //將 adapter 放入 ViewPager
        viewPager?.adapter = adapter

        //指定一開始 ViewPager 顯示內容位置
        viewPager?.currentItem = 0
    }
}